### PR TITLE
fix: Set timeouts only when values are given

### PR DESCRIPTION
**Pull request by dbast**
## Description

Fixes https://github.com/terraform-aws-modules/terraform-aws-lambda/issues/486

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -142,6 +142,12 @@ resource "aws_lambda_function" "this" {
     aws_iam_role_policy_attachment.vpc,
     aws_iam_role_policy_attachment.tracing,
   ]
+
+  lifecycle {
+    ignore_changes = [
```
Comment (antonbabenko) [inline main.tf:147]: There are better solutions than this.

Timeouts should be rewritten with `dynamic`, like:

```hcl
  dynamic "timeouts" {
    for_each = length(var.timeouts) > 0 ? [true] : []

    content {
      create = try(var.timeouts.create, null)
      update = try(var.timeouts.update, null)
      delete = try(var.timeouts.delete, null)
    }
  }
```

Please update this PR and test it.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (dbast) [reply main.tf]: Done

**Comment by dbast** [comment]
@antonbabenko thanks... rebased, changed and tested via internal terraform enterprise deployment.... ready for another review.

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 6.7.1 :tada:

**Pull request merged**